sos/cleaner/__init__.py

```diff
@@ -612,28 +612,48 @@ def obfuscate_file(self, filename, short_name=None, arc_name=None):
         if not filename:
             # the requested file doesn't exist in the archive
             return
-        self.log_debug("Obfuscating %s" % short_name or filename,
-                       caller=arc_name)
         subs = 0
-        tfile = tempfile.NamedTemporaryFile(mode='w', dir=self.tmpdir)
-        with open(filename, 'r') as fname:
-            for line in fname:
-                try:
-                    line, count = self.obfuscate_line(line)
-                    subs += count
-                    tfile.write(line)
-                except Exception as err:
-                    self.log_debug("Unable to obfuscate %s: %s"
-                                   % (short_name, err), caller=arc_name)
-        tfile.seek(0)
-        if subs:
-            shutil.copy(tfile.name, filename)
-        tfile.close()
-        _ob_filename = self.obfuscate_string(short_name)
+        if not os.path.islink(filename):
+            # don't run the obfuscation on the link, but on the actual file
+            # at some other point.
```

Comment:
Maybe my bad understanding of the English, but here we do obfuscate the real file content, not the symlink. While I read the comment "run obfuscation on the actual file at some other point, not now"..?

Reply:
From `Archive.get_file_list()` we'll end up iterating over _both_ the link and the real file the link points to. Since there are potentially 2 filenames to obfuscate (both real and the link), skip the real file content obfuscation until our iteration hits the real file in the file list.